src/schema/partner.js

```diff
@@ -247,6 +247,9 @@ const PartnerType = new GraphQLObjectType({
             partner_id: _id,
           }).then(response => {
             if (response.errors) {
+              if (response.errors.toString().match(/Document not found/)) {
```

Comment:
This error message _could_ conceivably change in the future upstream (in Lewitt), maybe it should use: https://github.com/artsy/lewitt-api/blob/7992fa5fb21a1f7a1481b5ed0a2f285181337d64/app/api/graph/root_query.rb#L11 (status code).

The 404 status code seems more future-proof (I love the idea of adding it in to the metadata there).

Re: testing, I guess i would see if throwing an error from https://github.com/artsy/metaphysics/blob/36fe4443fd79f9ae3bfd80ec5ca79a418f5d432c/src/schema/__tests__/partner.test.js#L85-L87 gives you anything to work with test-wise, but yea its a tricky test!

Reply:
> The 404 status code seems more future-proof (I love the idea of adding it in to the metadata there).

Yeah the idea of adding HTTP status codes was that clients could implement stable logic, rather than e.g. string matching on the error message.